crates/polars-plan/src/logical_plan/hive.rs

```diff
@@ -115,3 +95,51 @@ impl HivePartitions {
             .collect()
     }
 }
+
+/// Convert a Hive partition string (e.g. "column=1.5") to a single-value [`Series`].
+fn hive_string_to_series(part: &str, schema: Option<SchemaRef>) -> Option<Series> {
+    let mut it = part.split('=');
+    let name = it.next()?;
+    let value = it.next()?;
+
+    // Having multiple '=' doesn't seem like a valid Hive partition.
+    if it.next().is_some() {
+        return None;
+    }
+
+    // Files are not Hive partitions, so globs are not valid.
+    if value.contains('*') {
+        return None;
+    }
+
+    let dtype = match schema {
+        Some(ref s) => s.get(name),
+        None => None,
+    };
+
+    value_to_series(name, value, dtype)
+}
+
+fn value_to_series(name: &str, value: &str, dtype: Option<&DataType>) -> Option<Series> {
+    let mut s = if INTEGER_RE.is_match(value) {
```

Comment:
Not for this PR. But I want to re-use the static regexes we use in the csv parser. As it is now, we have multiple statics doing the same thing (hopefully).

Reply:
Will add it to the tracking issue!